src/Microsoft.Health.Operations.Functions/Management/PurgeOrchestrationInstanceHistory.cs

```diff
@@ -42,6 +45,7 @@ public PurgeOrchestrationInstanceHistory(IOptions<PurgeHistoryOptions> options)
         PurgeHistoryOptions value = EnsureArg.IsNotNull(options?.Value, nameof(options));
         _statuses = EnsureArg.HasItems(value.Statuses, nameof(options));
         _minimumAge = TimeSpan.FromDays(EnsureArg.IsGt(value.MinimumAgeDays, 0, nameof(options)));
+        _instancesToSkipPurging = value.InstancesToSkipPurging == null ? new HashSet<string>() : value.InstancesToSkipPurging.ToHashSet<string>();
```

Comment:
While often a good idea to use a `HashSet<string>` I can't imagine these collections are going to be large enough to merit it

Reply:
Changed to array